order/src/main/java/com/yas/order/service/CheckoutService.java

```diff
@@ -33,70 +35,94 @@
-            .stream()
-            .map(checkoutItemPostVm -> {
-                CheckoutItem item = checkoutMapper.toModel(checkoutItemPostVm);
-                item.setCheckoutId(savedCheckout.getId());
-                return item;
-            })
-            .toList();
+        CheckoutVm checkoutVm = checkoutMapper.toVm(checkout);
+        Set<CheckoutItemVm> checkoutItemVms = checkout.getCheckoutItems()
+                .stream()
+                .map(checkoutMapper::toVm)
+                .collect(Collectors.toSet());
 
-        List<CheckoutItem> savedCheckoutItems = checkoutItemRepository.saveAll(checkoutItemList);
+        return checkoutVm.toBuilder().checkoutItemVms(checkoutItemVms).build();
+    }
 
-        Set<CheckoutItemVm> checkoutItemVms = savedCheckoutItems
-            .stream()
-            .map(checkoutMapper::toVm)
-            .collect(Collectors.toSet());
+    private Checkout setCheckoutItems(Checkout checkout, CheckoutPostVm checkoutPostVm) {
+        Set<Long> productIds = new HashSet<>();
+        List<CheckoutItem> checkoutItems = checkoutPostVm.checkoutItemPostVms()
+                .stream()
+                .map(checkoutItemPostVm -> {
+                    CheckoutItem item = checkoutMapper.toModel(checkoutItemPostVm);
+                    item.setCheckout(checkout);
+                    productIds.add(item.getProductId());
+                    return item;
+                })
+                .toList();
+
+        Map<Long, ProductCheckoutListVm> products = 
+                productService.getProductInfomation(productIds, 0, productIds.size());
+        associateProductWithCheckoutItem(checkout, products, checkoutItems);
+
+        checkout.setCheckoutItems(checkoutItems);
+        return checkout;
+    }
 
-        return checkoutVm.toBuilder().checkoutItemVms(checkoutItemVms).build();
+    private void associateProductWithCheckoutItem(
+            Checkout checkout,
+            Map<Long, ProductCheckoutListVm> products,
+            List<CheckoutItem> checkoutItems) {
+        checkoutItems.forEach(t -> {
+            ProductCheckoutListVm product = products.get(t.getProductId());
+            if (product == null) {
+                throw new NotFoundException("PRODUCT_NOT_FOUND", t.getProductId());
+            } else {
+                t.setProductName(product.getName());
+                t.setProductPrice(BigDecimal.valueOf(product.getPrice()));
+                checkout.addAmount(product.getPrice() * t.getQuantity());
+            }
+        });
     }
 
     public CheckoutVm getCheckoutPendingStateWithItemsById(String id) {
 
         Checkout checkout = checkoutRepository.findByIdAndCheckoutState(id, CheckoutState.PENDING).orElseThrow(()
-            -> new NotFoundException(CHECKOUT_NOT_FOUND, id));
+                -> new NotFoundException(CHECKOUT_NOT_FOUND, id));
 
         if (!checkout.getCreatedBy().equals(AuthenticationUtils.getCurrentUserId())) {
```

Comment:
As an ADMIN user, if I am not the user who created the checkout, would I be able to call this method?
Please extract this validation to a separate method

Reply:
No, I don't think you should. This API is to view checkouts of a user account.
To view checkouts of a user as an ADMIN user, there should be another backoffice get-all-with-parameters API